src/main/java/max/data/PersistentDataHandler.java

```diff
@@ -0,0 +1,225 @@
+package max.data;
+
+import max.task.*;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class PersistentDataHandler {
+    // TODO refactor out the tokenizers into a seperate class
+
+    private static final String delimiter = " -- ";
+
+    // These constants define the index position of task arguments in the token array
+    private static final int TASK_TYPE_INDEX = 0;
```

Comment:
Good use of constants to avoid magic numbers! Keep this up for the rest of your code!